Tests/Core/NetworkObserverTests.swift

```diff
@@ -80,4 +80,63 @@ class NetworkObserverTests: OperationTests {
             XCTAssertFalse(self.visibilityChanges[1])
         }
     }
+
+    func test__network_indicator_changes_once_when_multiple_operations_start() {
+
+        let operation1 = TestOperation(delay: 1)
+        operation1.addObserver(NetworkObserver(indicator: indicator))
+
+        let operation2 = TestOperation(delay: 1)
+        operation2.addObserver(NetworkObserver(indicator: indicator))
+
+        addCompletionBlockToTestOperation(operation1, withExpectation: expectationWithDescription("Test: \(#function)"))
+        addCompletionBlockToTestOperation(operation2, withExpectation: expectationWithDescription("Test: \(#function)"))
```

Comment:
What is the purpose / need behind having these "completion" blocks (really `DidFinishObserver`)? I don't understand the timing behind when these expectations get fulfilled. The tests test for the operation having finished, but the `visbilityChanges` don't include the changes that would happen with the `NetworkObserver`.

It seems like ideally these tests would be able to properly tests for visibility changes starting _and_ stopping. And you wouldn't need two different tests to test that functionality.